### ISSUE TITLE
Loss of accuracy of Flan-T5-Large when using default converted model

### ISSUE THREAD

**Issue by Matthieu-Tinycoaching** [OpenNMT/CTranslate2#1066]
Hi,

I have converted the `google/flan-t5-large` model with the following command:
```
model_id = "google/flan-t5-large"
HF_tc = TransformersConverter(model_name_or_path=model_id)
HF_tc.convert(output_dir="./flan-t5-large/default")
```
Then, I use the following code to answer to a question based on a context:
```
model_path = "./flan-t5-large/default"

context = "Le Père Noël (né Thomas Patrick O'Connor ; mai 1947)[1] est un homme politique américain, un moine et un militant de la protection de l'enfance[2]. Il est membre du conseil municipal de North Pole, en Alaska, depuis 2015[3]. Claus a été réélu au conseil en 2019[4], avec 100 voix[5]. Il est actuellement le maire pro tem[6]. Il a été candidat à la Chambre des représentants des États-Unis lors de l'élection spéciale de 2022. Claus est un moine chrétien de l'ordre anglican celtique Anam Cara[8]. Il a changé son nom pour correspondre au personnage légendaire en 2005, afin d'aider à faire connaître son activisme pour la santé et le bien-être des enfants[9]. Claus est un partisan du socialisme démocratique, y compris le soutien à l'assurance-maladie pour tous, l'aide aux coronavirus, un impôt sur la fortune et l'annulation des prêts étudiants[10]."

question = "Pourquoi Claus a-t-il changé de nom ?"

input_text = f"Given the context please answer the question. Use only French words. Context: {context}; Question: {question}; Answer:"

translator = ctranslate2.Translator(model_path, device="cuda")
tokenizer = transformers.AutoTokenizer.from_pretrained(model_id)

input_tokens = tokenizer.convert_ids_to_tokens(tokenizer.encode(input_text))

results = translator.translate_batch(source=[input_tokens])

output_tokens = results[0].hypotheses[0]
output_text = tokenizer.decode(tokenizer.convert_tokens_to_ids(output_tokens))

print(output_text)
```
which generates the following text : `pour correspondre à sa a et l'annulation des prêts étudiants[10].Il a fait d'aider à faire à faire connaître son activisme pour la santé et le bien-être des enfants`

This output text is not valid nor correct and differs from the answer obtained with the transformers library, which is valid and correct: `Il a changé son nom pour correspondre au personnage légendaire en 2005, afin d'aider à faire connaître son activisme pour la santé et le bien-être des enfants[9].`

Are there any specific operations during conversion that may explains this loss of accuracy compared to the original model?

**Comment by guillaumekln** [comment]
Thanks for the details. It looks like there is an issue when the model is running on GPU. I will look into that.

Note that we get the expected output when running on CPU and setting `beam_size=1` in `translate_batch` (this is the default value in transformers' `model.generate`).

**Comment by guillaumekln** [comment]
This was a pretty basic mistake which will be fixed by the PR above. I guess I only tested the "small" version, but the bug shows in "base" and larger versions.

**Comment by Matthieu-Tinycoaching** [comment]
Nice @guillaumekln ! Do you have an ETA of the next release to test it?

**Comment by guillaumekln** [comment]
I will try to make a new version this week. In the meantime, you can grab the latest package from the [build page](https://github.com/OpenNMT/CTranslate2/actions/runs/4116424471). Download the "python-wheels" artifact, extract the archive, and install the appropriate Python wheel.

**Comment by guillaumekln** [comment]
FYI, I just released version 3.5.0 which includes the fix.
